components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1488,20 +1557,63 @@ export function createWalletApi (
                     }
 
                     return acc
-                  }, {})
+                  }, baseTokenBalances)
                 }
               } else {
+                console.error(
+                  `Error calling jsonRpcService.getSPLTokenBalances:
+                  error=${result.errorMessage}
+                  arg=`, arg
+                )
+              }
+            }
+
+            // Fallback to fetching individual balances
+            const tokens = (arg.tokens ?? [])
+              .filter(token => !isNativeAsset(token))
+
+            let balances = baseTokenBalances
+            for (let i = 0; i < tokens.length; i += 10) {
+              const chunk = tokens.slice(i, i + 10)
+              const chunkBalances = await mapLimit(chunk, 10, async (token: BraveWallet.BlockchainToken) => {
```

Comment:
Not sure if this actually limiting anything here since we as setting max concurrency to 10, with a chink size of 10. Loosk like the chunking is what is actually limiting the max amount of promise creations

Reply:
Damn! That's a nice catch. Fixed it here: https://github.com/brave/brave-core/pull/18735/commits/1d7aeab8769c5ca2be5cd640149c41ead2a3fcb4